lib/model/store.dart

```diff
@@ -171,6 +172,26 @@ abstract class GlobalStore extends ChangeNotifier {
   String toString() => '${objectRuntimeType(this, 'GlobalStore')}#${shortHash(this)}';
 }
 
+/// A wrapper class for `Map<K, V>`.
+///
+/// This can be listened to for any modifications done to `Map<K, V>`.
+class MapNotifier<K, V> extends ValueNotifier<Map<K, V>> {
+  MapNotifier(super._value);
```

Comment:
The reason for this change is that `users` used in `model/autocomplete.dart` are now sorted by relevance before being filtered. Before this PR, an `iterator` was made from `users.values` which kept the relationship with `users` so any modification to `users` would be reflected in the iterator too, for example, `ConcurrentModificationError` being thrown. For this change, `users` should be changed to a list using `users.values.toList` in order to be sorted, and this breaks the relationship with `users`. By using this wrapper, we can listen to modifications to the original `users` and act upon that.

Reply:
I see, that makes sense.

Definitely belongs in its own prep commit, as discussed in https://github.com/zulip/zulip-flutter/pull/608#discussion_r1592958809. :slightly_smiling_face: It's fine that as of that commit, the new functionality won't really be *needed*; the commit message can explain why we're about to start needing it.

Reply:
Looking at the code, adding, removing, or updating a user will start the search again. But I added this specific `notifier` to pass a failing test in `autocomplete_test.dart` ("MentionAutocompleteView mutating store.users while in progress causes retry").